src/main/java/Spiderman.java

```diff
@@ -0,0 +1,212 @@
+                    taskList.add(new Event(description, from, to));
+                }
+                catch (DateTimeParseException e) {
+                    System.out.println("The date and time is not in the correct format!");
+                    continue;
+                }
+                catch (Exception e) {
+
+                    System.out.println("The from and/or to cannot be empty!");
+                    continue;
+                }
+                System.out.println("Cool! I'll add this to your task list!");
+                System.out.println("You now have " + taskList.size() + " tasks in your task list.");
+            }
+            else if (splitInput[0].equals("todo")) {
+                String description = splitInputByCommand[0].replaceFirst("todo", "").trim();
+                if (description.isEmpty()) {
+                    System.out.println("The description of a todo cannot be empty.");
+                    continue;
+                }
+
+                taskList.add(new Todo(description));
+                System.out.println("Cool! I'll add this to your task list!");
+                System.out.println("You now have " + taskList.size() + " tasks in your task list.");
+            }
+            else {
+                System.out.println("Sorry, I do not understand what you mean. Check the README file for the list of known actions!");
+            }
+        }
+
+        saveTasksToFile(taskList);
+
+        // Exit program message
+        System.out.println("Bye. Hope to see you again soon!");
+        scan.close();
+    }
+
+    // Method to save tasks to file
+    private static void saveTasksToFile(ArrayList<Task> taskList) {
+        try (FileWriter writer = new FileWriter("tasks.txt")) {
+            for (Task task : taskList) {
+                writer.write(task.toTextFormat());
+                writer.write("\n");
+            }
+            System.out.println("Tasks have been saved to tasks.txt.");
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving tasks to file.");
+        }
+    }
+
+    private static ArrayList<Task> loadTasksFromFile() {
+        ArrayList<Task> taskList = new ArrayList<>();
+        try {
+            File myObj = new File("tasks.txt");
+            Scanner scan = new Scanner(myObj);
+            while (scan.hasNextLine()) {
+                String data = scan.nextLine();
+                String[] savedTasks = data.split("\\|");
+                // If task is a todo
+                if (savedTasks[0].equals("T")) {
+                    Todo task = new Todo(savedTasks[2]);
+                    if (savedTasks[1].equals("T")) {
+                        task.markAsDone();
+                    } else {
+                        task.markAsNotDone();
+                    }
+                    taskList.add(task);
+                }
+
+                // If task is a deadline
+                if (savedTasks[0].equals("D")) {
```

Comment:
In the loadTasksFromFile method, the if statements for checking task types (e.g., if (savedTasks[0].equals("T"))) could be refactored to improve clarity and reduce redundancy. Consider using a switch statement based on savedTasks[0] to handle different task types. This approach will make the code more readable and easier to manage as more task types are added in the future!